vc_zoom/indico_vc_zoom/client/JoinButton.tsx

```diff
@@ -0,0 +1,150 @@
+// This file is part of the Indico plugins.
+// Copyright (C) 2020 - 2024 CERN and ENEA
+//
+// The Indico plugins are free software; you can redistribute
+// them and/or modify them under the terms of the MIT License;
+// see the LICENSE file for more details.
+
+import staticURL from 'indico-url:plugin_vc_zoom.static';
+
+import React, {useState} from 'react';
+import {Button, ButtonGroup, Confirm, Icon, Label, Popup, SemanticICONS} from 'semantic-ui-react';
+
+import {Translate} from 'indico/react/i18n';
+import {handleAxiosError, indicoAxios} from 'indico/utils/axios';
+
+import './JoinButton.module.scss';
+
+interface OptionsButtonProps {
+  url: string;
+  onMadeAltHost: () => void;
+}
+
+/** A dropdown button which shows additional actions, such as the possibility to take over as meeting co-host */
+function OptionsButton({url, onMadeAltHost}: OptionsButtonProps) {
+  const [isConfirmOpen, setConfirmOpen] = useState(false);
+  const [state, setState] = useState('idle');
+
+  async function makeAlternativeHost() {
+    setConfirmOpen(false);
+    try {
+      setState('submitting');
+      await indicoAxios.post(url);
+      setState('success');
+    } catch (error) {
+      handleAxiosError(error);
+    }
+    window.setTimeout(() => {
+      onMadeAltHost();
+      setState('idle');
+    }, 3000);
+  }
+
+  const trigger = (
+    <Button
+      disabled={state !== 'idle'}
+      title={Translate.string('Actions you can perform on your Zoom meeting')}
+      color={state === 'success' ? 'green' : undefined}
+      loading={state === 'submitting'}
+      size="mini"
+      icon={state !== 'submitting'}
+    >
+      {
+        {
+          idle: <Icon fitted name="cog" />,
+          submitting: null,
+          success: <Icon fitted name="checkmark" />,
+        }[state]
+      }
+    </Button>
+  );
+
+  return (
+    <>
+      <Popup trigger={trigger} hoverable on="click">
```

Comment:
Did you actually implement that with SemanticUI React or it's just a mockup? Because turning that into a Dropdown breaks the CSS (as the button is nested into the dropdown itself).

Reply:
Done.